pkg/engine/k8smanifest.go

```diff
@@ -0,0 +1,355 @@
+	// load AdmissionRequest
+	request, err := policyContext.JSONContext.Query("request")
+	if err != nil {
+		return false, "", fmt.Errorf("failed to get a request from policyContext: %s", err.Error())
+	}
+	reqByte, _ := json.Marshal(request)
+	var adreq *admissionv1.AdmissionRequest
+	err = json.Unmarshal(reqByte, &adreq)
+	if err != nil {
+		return false, "", fmt.Errorf("failed to unmarshal a request from requestByte: %s", err.Error())
+	}
+	// unmarshal admission request object
+	var resource unstructured.Unstructured
+	objectBytes := adreq.Object.Raw
+	err = json.Unmarshal(objectBytes, &resource)
+	if err != nil {
+		return false, "", fmt.Errorf("failed to Unmarshal a requested object: %s", err.Error())
+	}
+
+	logger.V(4).Info("verifying manifest...", adreq.Namespace, adreq.Kind.Kind, adreq.Name, adreq.UserInfo.Username)
+
+	// allow dryrun request
+	if adreq.DryRun != nil && *adreq.DryRun {
+		return true, "allowed because of DryRun request", nil
+	}
+
+	// prepare verifyResource option
+	vo := &k8smanifest.VerifyResourceOption{}
+	// adding default ignoreFields from
+	// github.com/sigstore/k8s-manifest-sigstore/blob/main/pkg/k8smanifest/resources/default-config.yaml
+	vo = k8smanifest.AddDefaultConfig(vo)
+	// adding default ignoreFields from pkg/engine/resources/default-config.yaml
+	vo = addDefaultConfig(vo)
+	// adding ignoreFields from Policy
+	for _, i := range verifyRule.IgnoreFields {
+		converted := k8smanifest.ObjectFieldBinding(i)
+		vo.IgnoreFields = append(vo.IgnoreFields, converted)
+	}
+
+	// dryrun setting
+	vo.DisableDryRun = !verifyRule.DryRunOption.Enable
+	if verifyRule.DryRunOption.Namespace != "" {
+		vo.DryRunNamespace = verifyRule.DryRunOption.Namespace
+	} else {
+		vo.DryRunNamespace = config.KyvernoNamespace()
+	}
+
+	// can be overridden per Attestor
+	if verifyRule.Repository != "" {
+		vo.ResourceBundleRef = verifyRule.Repository
+	}
+
+	// signature annotation
+	// set default annotation domain
+	if verifyRule.AnnotationDomain != "" && verifyRule.AnnotationDomain != DefaultAnnotationKeyDomain {
+		vo.AnnotationConfig.AnnotationKeyDomain = verifyRule.AnnotationDomain
+	}
+
+	// signature verification by each attestor
+	verifiedMsgs := []string{}
+	for i, attestorSet := range verifyRule.Attestors {
+		path := fmt.Sprintf(".attestors[%d]", i)
+		verified, reason, err := verify(resource, attestorSet, vo, path, string(adreq.UID), logger)
+		if err != nil {
+			return verified, reason, err
+		}
+		if !verified {
+			return verified, reason, err
+		} else {
+			verifiedMsgs = append(verifiedMsgs, reason)
+		}
+	}
+	msg := fmt.Sprintf("verified manifest signatures; %s", strings.Join(verifiedMsgs, ","))
+	return true, msg, nil
+}
+
+func verify(resource unstructured.Unstructured, attestorSet kyvernov1.AttestorSet, vo *k8smanifest.VerifyResourceOption, path string, uid string, logger logr.Logger) (bool, string, error) {
```

Comment:
Is it possible to simplify this function and try to reuse logic from ` verifyAttestorSet` 
 https://github.com/kyverno/kyverno/blob/main/pkg/engine/imageVerify.go#L309?